src/FileSystem/FilesFinder.php

```diff
@@ -70,7 +74,7 @@ function (string $file): bool {
 
         // filtering files in directories collection
         $directories = $this->fileAndDirectoryFilter->filterDirectories($filesAndDirectories);
-        $filteredFilePathsInDirectories = $this->findInDirectories($directories, $suffixes, $sortByName);
+        $filteredFilePathsInDirectories = $this->findInDirectories($directories, $suffixes, $sortByName, $onlySuffix);
```

Comment:
This should check on `$filteredFilePaths` variable as well.

I am currently on mobile, i will check when I back to PC.

Reply:
I don't think that's necessary, as explicit files should be always used. That's the behavior I'd expect :)

Reply:
The `files` can be registered via `withPaths()`, which means unrelated to the `--only-suffix` use, show invalid expectation, eg `--only-suffix=controllers`, but `rector.php` and `.php_cs_fixer.php` also changed.

It should either check for both `$filteredFilePaths` and `$filteredFilePathsInDirectories`, or filter in final collection files result.

I will create new PR with what I mean.

Reply:
see PR:

- https://github.com/rectorphp/rector-src/pull/6648

for include other files unrelated to the `--ony-suffix` in the configration of paths provided.